src/nplinker/scoring/object_link.py

```diff
@@ -1,3 +1,15 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+
+if TYPE_CHECKING:
```

Comment:
Why the imports happen only if TYPE_CHECKING is true?

Reply:
Because they are only used for static typing check (.i.e. running mypy), it's not necessary to import them for actually executing code.

Reply:
oh yes sorry I see that indeed they're never actually used in the code, except for type hints